test/functional/autocmd/termxx_spec.lua

```diff
@@ -90,6 +90,19 @@ describe('autocmd TermClose', function()
     retry(nil, nil, function() eq('3', eval('g:abuf')) end)
     feed('<c-c>:qa!<cr>')
   end)
+
+  it('exposes v:event.status', function()
+    nvim('set_option', 'shell', nvim_dir .. '/shell-test')
+    command('autocmd TermClose * let g:status = v:event.status')
+    command('terminal')
+    retry(nil, nil, function() neq(-1, eval('jobwait([&channel], 0)[0]')) end)
+    eq(0, eval('g:status'))
+
+    nvim('set_option', 'shellcmdflag', 'EXIT')
+    command('terminal 42')
+    retry(nil, nil, function() neq(-1, eval('jobwait([&channel], 0)[0]')) end)
+    eq(42, eval('g:status'))
```

Comment:
windows CI is failing, but I think this might be a race, maybe TermClose doesn't fire until after this eval(). To avoid it (also makes the code shorter), use retry() on this eval(). The other retry() statements can be deleted.

    retry(nil, nil, function() neq(42, eval('g:status')) end)

Reply:
Still failing after this change. I had to modify the shell-test program in order to make it trigger an error code: https://github.com/neovim/neovim/pull/15406/files#diff-d628f396520ec980c103d1dd0938709a09d5cd1d9164187cb6282f11f9508431R127-R132

Any chance this could be giving us problems on Windows? It looks like it's just returning error code 1 no matter what.

Reply:
do we need wmain() with wchar_t? that is what this other test fixture does: https://github.com/neovim/neovim/blob/2ae9ff128583984145ce38e61bbe9047871616bf/test/functional/fixtures/printenv-test.c#L13

Reply:
wmain() is probably not needed and will require a lot of code. Also this test works on Windows CI, which also is using sscanf() in shell-test.c: https://github.com/neovim/neovim/blob/29095b2a36babd0fc84823b2ecb60ed738d06dd7/test/functional/ui/wildmode_spec.lua#L117